src/json_printer.sql

```diff
@@ -55,40 +70,65 @@ package body "JSON_PRINTER" as
       return str;
     end if;
   end llcheck;  
-
-  function escapeString(str varchar2) return varchar2 as
-    sb varchar2(32767) := '';
-    buf varchar2(40);
-    num number;
+  
+  -- escapes a single character. 
+  function escapeChar(ch char) return varchar2 deterministic is
```

Comment:
I personally belive that 
   Return X;
Is faster than 
  Y:=X;
   Return Y;
 Of course the optimizer COULD do its magic and rewrite, under the hood,  the code as i wrote it, but I . But *personally* I find it more readable this way: I don't like procedures that force me to read the whole code until the last line just to discover that nothing happens to the intermediate Y variable,  until the single return statement. 

 Anyway,  feel free to refactor the code as you prefer.  The point of the optimization is the use of the associative array. The rest is just cosmetic changes i made to make it more readable (for my eyes)